capeinfra/swimlanes/private.py

```diff
@@ -53,3 +57,173 @@ def default_cfg(self) -> dict:
             },
             "private-subnets": [],
         }
+
+    def create_dap_api(self):
+        """Create the data analysis pipeline API for the private swimlane."""
+
+        # TODO: as this will be an api, there will be a number of functions
+        #       most likely. we may want to consider making an api class that
+        #       does some setup in a general manner and holds onto a mapping
+        #       or list of functions eventually. this will almost certainly
+        #       not be our only api and a reusable pattern would be nice.
+
+        # TODO: we can probably have one lambda role for the whole API (or even
+        #       multiple apis), we don't want to make one for each endpoint
+        #       probably...
+        self.api_lambda_role = get_inline_role(
+            f"{self.basename}-dapapi-lmbd-role",
+            f"{self.desc_name} data analysis pipeline lambda role",
+            "lmbd",
+            "lambda.amazonaws.com",
+            # TODO: for right this second, the lambda functions needs no extra
+            #       permissions via a role policy (the toy function currently
+            #       only logs that it was called). This will almost certainly
+            #       change, at which point a new function should be made in
+            #       iam.py to build the policy doc. It is likely that this
+            #       lambda would write to a queue, dynamodb, or otherwise put
+            #       the params of the DAP into a holding place for the head
+            #       node scheduler to get it going eventually. Until then, the
+            #       role_policy param will be None
+            None,
+            "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
```

Comment:
```suggestion
            # TODO: for right this second, the lambda functions needs no extra
            #       permissions via a role policy (the toy function currently
            #       only logs that it was called). This will almost certainly
            #       change, at which point a new function should be made in
            #       iam.py to build the policy doc. It is likely that this
            #       lambda would write to a queue, dynamodb, or otherwise put
            #       the params of the DAP into a holding place for the head
            #       node scheduler to get it going eventually. Until then, the
            #       role_policy param will be None
            srvc_policy_attach="arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
```

because there is a default value, we don't need to specify this `None`

Reply:
this was intentional. i left it None because I know it'll change and just wanted to overwrite the word instead of remember to add a new line and comma (hence the explicit mention of None at the end of the comment)...